src/makerom/rom_header.h

```diff
@@ -0,0 +1,143 @@
+#ifndef ROM_HEADER_H
+#define ROM_HEADER_H
+
+/* Storage medium IDs, used internally in MEDIUM below */
+
+#define STORAGE_MEDIUM_CARTRIDGE            "N"
+#define STORAGE_MEDIUM_CARTRIDGE_EXPANDABLE "C"
+#define STORAGE_MEDIUM_DISK                 "D"
+#define STORAGE_MEDIUM_DISK_EXPANSION       "E"
+
+/* Region IDs, used internally in REGION below */
+
+#define REGION_CODE_ALL      "A"
+#define REGION_CODE_JP       "J"
+#define REGION_CODE_US       "E"
+#define REGION_CODE_PAL      "P"
+#define REGION_CODE_GATEWAY  "G"
+#define REGION_CODE_LODGENET "L"
+
+/**
+ * Magic value to determine if the ROM is byteswapped.
+ *
+ * This is not totally reliable since the PI Domain 1 Latency could also hold this value, however generally it can be
+ * assumed that this is not the case.
+ */
+#define ENDIAN_IDENTIFIER \
+    .byte 0x80
+
+/**
+ * Configures the timings for PI Domain 1. This determines how fast the PI hardware will read from the ROM. IPL2 reads
+ * this configuration using the slowest possible configuration before switching to the provided configuration.
+ */
+#define PI_DOMAIN_1_CFG(lat, pwd, pgs, rls)   \
+    .byte (((rls) & 3) << 4) | ((pgs) & 0xF); \
+    .byte (pwd) & 0xFF;                       \
+    .byte (lat) & 0xFF
+
+/**
+ * Some older libultra versions use this field to set osClockRate. It does not have any other meaningful effect, the
+ * clock rate of physical units does not actually change to match this value.
+ */
+#define SYSTEM_CLOCK_RATE_SETTING(num) \
+    .word (num)
+
+/**
+ * Indicates the entrypoint address of the program that IPL3 will load the boot segment to and execute.
+ * This should be >= 0x80000400.
+ */
+#define ENTRYPOINT(sym) \
+    .word (sym)
+
+/**
+ * Indicates the hardware revision the program is designed for (hw_major, hw_minor) and what libultra version (os_ver)
+ * it uses.
+ * The hardware revision for a retail N64 is (2,0).
+ * The libultra version may be a single letter, without quotes.
+ */
```

Comment:
Im assuming the formatter did these line breaks?
You can coax block comments into some better line breaks by just adding your own. For example for this case I would suggest something like 
```c
/**
 * Indicates the hardware revision the program is designed for (hw_major, hw_minor)
 * and what libultra version (os_ver) it uses.
 *
 * The hardware revision for a retail N64 is (2,0).
 * The libultra version may be a single letter, without quotes.
 */
 ```